packages/components/Tabs/src/TabsItem.tsx

```diff
@@ -0,0 +1,153 @@
+/** @jsx withSlots */
+import * as React from 'react';
+import { View } from 'react-native';
+import { compose, IUseComposeStyling } from '@uifabricshared/foundation-compose';
+import { ISlots, withSlots } from '@uifabricshared/foundation-composable';
+import { Text } from '@fluentui-react-native/text';
+import { Icon } from '@fluentui-react-native/icon';
+import { settings, tabsItemSelectActionLabel } from './TabsItem.settings';
+import { backgroundColorTokens, borderTokens, textTokens, foregroundColorTokens, getPaletteFromTheme } from '@fluentui-react-native/tokens';
+import { filterViewProps } from '@fluentui-react-native/adapters';
+import { mergeSettings } from '@uifabricshared/foundation-settings';
+import { TabsContext } from './Tabs';
+import { tabsItemName, TabsItemType, TabsItemProps, TabsItemSlotProps, TabsItemRenderData, TabsItemState } from './TabsItem.types';
+import {
+  useAsPressable,
+  useViewCommandFocus,
+  createIconProps,
+  useOnPressWithFocus,
+} from '@fluentui-react-native/interactive-hooks';
+
+export const TabsItem = compose<TabsItemType>({
+  displayName: tabsItemName,
+
+  usePrepareProps: (userProps: TabsItemProps, useStyling: IUseComposeStyling<TabsItemType>) => {
+    const defaultComponentRef = React.useRef(null);
+    const {
+      icon,
+      headerText = '',
+      accessibilityLabel = userProps.headerText,
+      componentRef = defaultComponentRef,
+      testID,
+      itemKey,
+      itemCount,
+      accessibilityPosInSet,
+      accessibilitySetSize,
+      ...rest
+    } = userProps;
+
+    // Grabs the context information from Tabs (currently selected TabsItem and client's onTabsClick callback)
+    const info = React.useContext(TabsContext);
+
+    /* We don't want to call the user's onTabsClick multiple times on the same selection. */
+    const changeSelection = () => {
+      if (itemKey != info.selectedKey) {
+        info.onTabsClick && info.onTabsClick(itemKey);
+        info.getTabId && info.getTabId(itemKey, info.tabsItemKeys.findIndex(x => x == itemKey) + 1);
```

Comment:
We called it getTabId because getTabId is a callback function that is available for the users to overwrite via the tab's getTabId prop. The same goes for onTabsClick. In the changeSelection function we are just setting up the callback functions (onTabsClick and getTabId) with the necessary updated values  for the user to use.